yarn-project/simulator/src/public/enqueued_call_simulator.ts

```diff
@@ -283,48 +283,75 @@ export class EnqueuedCallSimulator {
       callContext: result.executionRequest.callContext,
       proverAddress: AztecAddress.ZERO,
       argsHash: computeVarArgsHash(result.executionRequest.args),
```

Comment:
Added the error messages below as the message thrown in padArray was not descriptive enough.